### PR TITLE
Includehttps

### PR DESCRIPTION
**Pull request by PepeluDev**
Hi @jgomezselles,

I have been using Hermes as traffic generator for a while and I thought that it would be awesome if it could use H2 and not only H2C. That is why I am creating this pull request.

This two changes make Hermes able to talk to HTTPS endpoints. Although it works, I am still working on the UTs for the HTTPS connection (managing self-signed certs and so) but I thought that it would be easier to start the conversation now. I will keep working on it and will make another pull request (or make a fix to the last commit, as you wish) once is done.

Cheers!

### CODE REVIEW

**Comment by jgomezselles** [comment]
Hi @PepeluDev 
Thanks a lot for the PR. It is indeed a very good (and needed) enhancement. While you finish the tests and, as requested, I'm starting a review. It's the first time I do it in github so I hope it goes well. Any feedback is welcome!

File: src/http2_client/client_impl.hpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ class client_impl : public client
 
     client_impl(std::shared_ptr<stats::stats_if> stats, boost::asio::io_context& io_ctx,
                 std::unique_ptr<traffic::script_queue_if> q, const std::string& h,
-                const std::string& p);
+                const std::string& p, bool secure_session = false);
```
Comment (jgomezselles) [inline src/http2_client/client_impl.hpp:50]:
```suggestion
                const std::string& p, const bool secure_session = false);
```

File: src/script/script_reader.cpp [hunk 2]
```diff
@@ -97,6 +97,15 @@ server_info script_reader::build_server_info()
...
     server.dns = get_value<std::string>("/dns");
     server.port = get_value<std::string>("/port");
+
+    if (is_present("/secure"))
+    {
+        server.secure = get_value<bool>("/secure");
+    } else
+    {
+        server.secure = false;
+    }
```
Comment (jgomezselles) [inline src/script/script_reader.cpp:107]: Minor: if you want, you could use:
```suggestion
    server.secure = is_present("/secure") ? get_value<bool>("/secure") : false;
```

**Review by jgomezselles — CHANGES_REQUESTED** [review]
Please, take a look at the comments.

- [x] I am proposing a way of not needing the shared_ptr
- [ ] Some bool parameters should be const
- [ ] Please, remember to run the format
- [ ] I think you'll need to change the manipulate function to get the tests working
- [ ] Waiting for those tests!

Anyhow, thanks a lot for implementing this. I really appreciate it.

**Comment by PepeluDev** [comment]
Hi again @jgomezselles thanks a lot for your review, I agree with all the changes. I will update the code asap and try to submit another revision, I still don't know how to fix commits in a PR, it's also my first PR. I will also update you with the UTs asap :)

**Comment by PepeluDev** [comment]
Hi @jgomezselles I have added 4 commits with the changes that you requested. Please take the time to review em whenever you are able and do not hesitate to ask for more! Now I will continue with the tests.
Cheers!!

**Comment by PepeluDev** [comment]
Hi again @jgomezselles, tests for secure connections are finally here. 

As a summary, to be able to launch a secure server that can be trusted by the client transparently, it is needed to have a trusted CA that can sign the server's certificate. The script "generateCryptographicMaterial" can do all the work, it generates the CA, updates the development container with the CA cert and generates the server's cryptographic material.

I took the freedom to modify a little bit the Dockerfiles, including the stuff that the script needs. I also included the command to run the tests when creating the final image, excluding the test "stats_test_extended.PrintAndWrite", which wasn't passing. Another consideration that you may take into account is that the step for downloading boost in the base image Dockerfile is not working anymore, I think they have changed the library location, see more [here](https://www.boost.org/users/download/).

**Comment by jgomezselles** [comment]
As discussed offline, I'm merging it (though it will break the CI). After that, I will make changes in a separate PR.
Thanks a lot for the contribution! :)

**Comment by PepeluDev** [comment]
Perfect then! I am happy to contribute to Hermes, it helped me a lot testing my services. I will keep an eye on the repo and will show it to my coworkers 😄

**Pull request merged**